### PR TITLE
[Enhancement] Image export error handling

### PR DESCRIPTION
**Pull request by heshan0131**
- Remove `charset=utf-8' from image data uri because it is not supported in all browsers
- Use `mini-svg-data-uri` to convert svg string to ascii
- Add catch error and display it in the notification panel, as well as image export modal

ezgif com-optimize (13)

### CODE REVIEW

File: src/components/modal-container.js [hunk 1]
```diff
@@ -301,16 +301,11 @@ export default function ModalContainerFactory(
             };
             break;
           case EXPORT_IMAGE_ID:
-            const { ratio, legend, resolution, exporting, imageDataUri } = uiState.exportImage;
             template = (
               <ExportImageModal
+                {... uiState.exportImage}
```
Comment (macrigiuseppe) [inline src/components/modal-container.js]: there is a space between __...__ and uiState.exportImage

File: src/components/notification-panel/notification-item.js [hunk 2]
```diff
@@ -88,29 +101,36 @@ export default function NotificationItemFactory()
     }
 
     render() {
-      const {notification, removeNotification} = this.props;
+      const {notification, removeNotification, isExpanded} = this.props;
```
Comment (macrigiuseppe) [inline src/components/notification-panel/notification-item.js]: isExpanded is not part of this component Props. In fact the property is part of the state. You can remove it

File: src/components/notification-panel/notification-item.js [hunk 3]
```diff
@@ -88,29 +101,36 @@ export default function NotificationItemFactory()
...
       return (
         <NotificationItemContent
           className="notification-item"
-          {...this.props}
+          type={notification.type}
+          isExpanded={isExpanded}
```
Comment (macrigiuseppe) [inline src/components/notification-panel/notification-item.js]: you can remove this as well

File: src/utils/dom-to-image.js [hunk 4]
```diff
@@ -194,6 +195,7 @@ function cloneNode(node, filter, root) {
 
   function makeNodeCopy(nd) {
     if (nd instanceof window.HTMLCanvasElement) {
+      console.log('window HTMLCanvasElement')
```
Comment (macrigiuseppe) [inline src/utils/dom-to-image.js]: remove log

File: src/utils/dom-to-image.js [hunk 5]
```diff
@@ -342,16 +344,20 @@ function makeSvgDataUri(node, width, height) {
...
+
+      const xhtml = util.escapeXhtml(serializedString);
+      const foreignObject = `<foreignObject x="0" y="0" width="100%" height="100%">${xhtml}</foreignObject>`;
+      const svgStr = `<svg xmlns="http://www.w3.org/2000/svg" width="${width}" height="${height}">${foreignObject}</svg>`;
+
+      // Optimizing SVGs in data URIs
+      // see https://codepen.io/tigt/post/optimizing-svgs-in-data-uris
+      // the best way of encoding SVG in a data: URI is data:image/svg+xml,[actual data].
+      // We don’t need the ;charset=utf-8 parameter because the given SVG is ASCII.
+      const dataUri = svgToMiniDataURI(svgStr);
```
Comment (macrigiuseppe) [inline src/utils/dom-to-image.js]: you can simply return the result of the __svgToMiniDataURI__

File: src/utils/dom-to-image.js [hunk 6]
```diff
@@ -470,7 +476,16 @@ function newUtil() {
       image.onload = () => {
         resolve(image);
       };
-      image.onerror = reject;
+      image.onerror = (err) => {
+        const message = `[kepler.gl] Failed to create image from data uri.
```
Comment (macrigiuseppe) [inline src/utils/dom-to-image.js]: define a constant for this one in constant file here: https://github.com/keplergl/kepler.gl/blob/master/src/constants/user-feedbacks.js

File: src/utils/dom-to-image.js [hunk 7]
```diff
@@ -470,7 +476,16 @@ function newUtil() {
...
+        Copy the uri in the javascript console when reporting this bug.
+        The uri is the string starts with "data:image/png"`;
+        console.warn(message, err);
```
Comment (macrigiuseppe) [inline src/utils/dom-to-image.js]: do we need this one?

File: src/utils/dom-to-image.js [hunk 8]
```diff
@@ -470,7 +476,16 @@ function newUtil() {
...
+        console.log(uri);
```
Comment (macrigiuseppe) [inline src/utils/dom-to-image.js]: this?

File: src/components/modal-container.js (hunk 1 above)
Comment (heshan0131) [reply src/components/modal-container.js]: fixed

File: src/components/notification-panel/notification-item.js (hunk 3 above)
Comment (heshan0131) [reply src/components/notification-panel/notification-item.js]: fixed

File: src/components/notification-panel/notification-item.js (hunk 2 above)
Comment (heshan0131) [reply src/components/notification-panel/notification-item.js]: fixed

File: src/utils/dom-to-image.js (hunk 5 above)
Comment (heshan0131) [reply src/utils/dom-to-image.js]: done

File: src/utils/dom-to-image.js (hunk 4 above)
Comment (heshan0131) [reply src/utils/dom-to-image.js]: done

File: src/utils/dom-to-image.js (hunk 8 above)
Comment (heshan0131) [reply src/utils/dom-to-image.js]: print the uri in console, so people can paste it while reporting this bug

File: src/utils/dom-to-image.js (hunk 6 above)
Comment (heshan0131) [reply src/utils/dom-to-image.js]: done

File: src/utils/dom-to-image.js (hunk 7 above)
Comment (heshan0131) [reply src/utils/dom-to-image.js]: removed

**Pull request merged**